fsspec/implementations/smb.py

```diff
@@ -138,9 +141,35 @@ def _connect(self):
                     encrypt=self.encrypt,
                     connection_timeout=self.timeout,
                 )
-                break
-            except Exception:
-                time.sleep(0.1)
+                return
+            except (
+                smbprotocol.exceptions.SMBAuthenticationError,
+                smbprotocol.exceptions.LogonFailure
+            ):
+                # These exceptions should not be repeated, as they clearly indicate
+                # that the credentials are invalid and not a network issue.
+                raise
+            except ValueError as exc:
+                if re.findall(r"\[Errno -\d+]", str(exc)):
+                    # This exception is raised by the smbprotocol.transport:Tcp.connect
+                    # and originates from socket.gaierror (OSError). These exceptions might
+                    # be raised due to network instability. We will retry to connect.
+                    retried_errors.append(exc)
+                else:
+                    # All another ValueError exceptions should be raised, as they are not
+                    # related to network issues.
+                    raise exc
+            except Exception as exc:
+                # Save the exception and retry to connect. This except might be dropped
+                # in the future, once all exceptions suited for retry are identified.
+                retried_errors.append(exc)
+
+            if retry < self.register_session_retries - 1:
+                time.sleep(1)
```

Comment:
May as well make the retry time configurable and ideally exponential, `base * factor**retry`

Reply:
That is great idea. Should the exponential option be also configurable? To avoid confusing a user?

Reply:
That's an interesting idea, but not the way around it is is other implementations ([example](https://github.com/fsspec/gcsfs/blob/main/gcsfs/retry.py#L129)). At least, then I would call it "max_wait_time", but probably better to have "base_wait_time" and backoff factor.